cloudinit/sources/__init__.py

```diff
@@ -221,6 +223,16 @@ def __init__(self, sys_cfg, distro, paths, ud_proc=None):
     def __str__(self):
         return type_utils.obj_name(self)
 
+    def _unpickle(self, ci_pkl_version: int) -> None:
```

Comment:
Does this need to be on `DataSource`? Could it instead be on `Paths` itself?

As background, the `networking` fix had to be in `Distro._unpickle` because affected `Networking` instances don't have any instance state: this means that they get unpickled differently, so the hooks we use don't get called. In this case, however, all `Paths` instances do have instance state, so the hooks we use _should_ get called.

Reply:
You are totally right here. I missed that the `Paths`  object can be instantiated directly. I will move the code there